src/Be.Vlaanderen.Basisregisters.Shaperon/DbaseBoolean.cs

```diff
@@ -0,0 +1,86 @@
+namespace Be.Vlaanderen.Basisregisters.Shaperon
+{
+    using System;
+    using System.IO;
+
+    public class DbaseBoolean : DbaseFieldValue
+    {
+        private bool? _value;
+
+        public DbaseBoolean(DbaseField field) : base(field)
+        {
+            if (field == null)
+                throw new ArgumentNullException(nameof(field));
+
+            if (field.FieldType != DbaseFieldType.Logical)
+                throw new ArgumentException(
+                    $"The field {field.Name}'s type must be logical to use it as a boolean field.", nameof(field));
+
+            if (field.DecimalCount.ToInt32() != 0)
+                throw new ArgumentException(
+                    $"The logical field {field.Name}'s decimal count must be 0 to use it as a boolean field.",
+                    nameof(field));
+
+            if (field.Length.ToInt32() != 1)
+                throw new ArgumentException(
+                    $"The logical field {field.Name}'s length must be 1 to use it as a boolean field.",
+                    nameof(field));
+
+            _value = default;
+        }
```

Comment:
Hm ... was too quick there. No, so, default is not going to do the right thing, because you want to initialize a `default(bool?)` whereas with this change causes `default(bool)` to be passed in. The same applies for all other datatypes.